docs/_static/link_gen/link.js

```diff
@@ -117,6 +117,18 @@ function changeTab(div) {
     }
 }
 
+/**
+ * Return name of directory git will clone given repo to.
+ *
+ * nbgitpuller needs to redirect users to *inside* the directory it
+ * just cloned. We copy the logic git itself uses to determine that.
+ * See https://github.com/git/git/blob/1c52ecf4ba0f4f7af72775695fee653f50737c71/builtin/clone.c#L276
+ */
+function generateCloneDirectoryName(gitCloneUrl) {
+    var lastPart = gitCloneUrl.split('/').slice(-1)[0];
+    return lastPart.split(':').slice(-1)[0].replace(/(\.git|\.bundle)?/, '');
+}
+
```

Comment:
Yikes, spent a while considering slice in python being confused until I realized this was not Python. Also got confused about the `:` separator as it won't be considered for a clone URL like `git@github.com:consideRatio/zero-to-jupyterhub-k8s.git` for example.

I conclude that the real crux is if we would ever do `new URL("git@myreposerver:myorg/myrepo.git"), because that would be invalid.

With this understanding, I assume this PR to not address the issue fully, as the following code snippet is still invalid.

```
var userName = new URL(repoUrl).pathname.split('/')[1];
```

I tried...

```
# this
new URL("https://github.com/consideRatio/zero-to-jupyterhub-k8s.git?query=test").pathname

# returns this
"/consideRatio/zero-to-jupyterhub-k8s.git"

# and this
new URL("https://github.com/consideRatio/zero-to-jupyterhub-k8s.git?query=test").pathname.split('/')[1]

# return this
consideRatio
```

That logic seems to be about generating a binder-link of some kind, where `userName` is extracted from the git link. I note that there are assumptions about this being GitHub repo specifically.

It seems its logic is already hardcoded towards GitHub and HTTP anyhow so there is no quick fix for that.

---

This LGTM.

Reply:
Yeah, this doesn't let binderhub use ssh clone URLs. But that was already the case, and I don't know if mybinder.org actually supports ssh URLs. I agree we should fix that, but not in this PR.